keps/sig-network/2681-pod-host-ip/README.md

```diff
@@ -0,0 +1,356 @@
+```
+
+### Maintaining Compatible Interworking between Old and New Clients
+
+[See the making a singular field plural](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#making-a-singular-field-plural)
+
+### Container Environment Variables
+
+The Downward API [status.hostIP](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api)
+will preserve the existing single IP address, and will be set to the default IP for each pod.
+A new pod API field named `status.hostIPs` will contain a list of IP addresses.
+The new pod API will have a slice of structures for the additional IP addresses. 
+Kubelet will translate the pod structures and return `status.hostIPs` as a comma-delimited string.
+
+Here is an example of how to define a pluralized `MY_HOST_IPS` environmental
+variable in a pod definition yaml file:
+
+``` yaml
+- name: MY_HOST_IPS
+  valueFrom:
+    fieldRef:
+      fieldPath: status.hostIPs
+```
+
+This definition will cause an environmental variable setting in the pod similar
+to the following:
+
+```
+# PodHostIPs FeatureGate is enabled
+MY_HOST_IPS=fd00:10:20:0:3::3,10.20.3.3
+
+# PodHostIPs FeatureGate is disabled
+MY_HOST_IPS=
+```
+
+### Test Plan
+
+Test whether FeatureGate behaves as expected when it is turned on or off
+Test whether Downward API supports `status.hostIPs`
+
+#### Expected behavior
+
+- If PodHostIPs FeatureGate is enabled:
+  - `status.hostIPs` there will be all host IPs (IPv4 and IPv6)
+- Else:
+  - `status.hostIPs` will be empty and the ApiServer will reject Downward API `status.hostIPs`, 
+    if Pods already existing Downward API `status.hostIPs`, ensure to ignore it and not affect others.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+  - Add `status.hostIPs` to the PodStatus API
+  - Downward API support for `status.hostIPs`
+- Basic units and e2e tests completed and enabled
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and users
+- Expand the e2e tests with more scenarios
+- Tests are in Testgrid and linked in the KEP
+
+#### Beta -> GA Graduation
+
+- 2 examples of end users using this field
+- Allowing time for feedback
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
```

Comment:
Please fill in FeatureGate name and component that will use it (only kube-apiserver? or do we need it in kubelet too maybe?)